### DISCUSSION TITLE
How to avoid State management and still manage global state in Next js

### DISCUSSION THREAD

**Discussion by krishnaacharyaa** [vercel/next.js#61326] [Help]
### Summary

![image](https://github.com/vercel/next.js/assets/116620586/1ad635d1-b2d7-4911-94d8-b176d0db947a)

I have application where A B C D are components, 
Lets assume we have state `stockLevelSelected` and this should be accesible to anyone and anyone should be able to mutate this.
For this the only solution i see if not maintaining a state in pass `setStockLevelSelectedHandler` and `stockLevelSelectedHandler` as props to the components which needs it. 
To avoid this I thought of using Context Provider, but this is in client side and it forces all of its children to be client side, so now what if I wanted to get api call on component C based on the `stockLevelSelected` which can be mutated by any other component like B A or D ?

API call would be done client side, how can i over come this ?

Is there different way of thinking on this.

Additionally I have initial data which is getting from the db, I want every component to have it when required, But also I want to revalidate the db, lets say in component D I want to revalidate, how do i do? because now Component D is client side.

### Additional information

I am open to any other optimizations on the github code provided below

### Example

https://github.com/krishnaacharyaa/nextjs-explore

**Comment by nirvikpurkait** [comment]
> it forces all of its children to be client side

In Next.Js you can use server-component inside of client-component, without making the the children server-component being forced to client. 

Maybe you are _importing_ the server-component in the client-component file, if you did so, insted of _importing_ the server-component in the client-component, pass it as a _prop_.

You can go through this [article](https://medium.com/@issam.ahw/demystifying-next-js-server-components-and-client-components-1b15ae405260#:~:text=To%20include%20a%20server%20component,client%20component%2C%20including%20server%20components.)

**Comment by krishnaacharyaa** [comment]
@nirvikpurkait Thank you for your answer, 
Even if i pass it as a server-component, I want to use context which requires client component. i.e
```
const { selectedDocument } = useAppContext(); // useAppContext needs "use client"
```

I want to fetch the `http://localhost:3000/api/summary/${selectedDocument?.DocumentId}` in server side  by using selectedDocument which i can get from potential client rendered component

**Reply by beshur** [reply]
But this one needs to be done inside a smaller client component, as low as possible down the component tree. I don't think there is a way around it.
